packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts

```diff
@@ -1,102 +1,239 @@
 import type { CSSObject } from 'styled-components';
-import type { BaseBoxProps } from './types';
-import { getIn, makeBorderSize, makeSize, makeSpace } from '~utils';
+import type {
+  BaseBoxProps,
+  MakeValueResponsive,
+  SpacingValueType,
+  ArrayOfMaxLength4,
+} from './types';
+import type { Breakpoints } from '~tokens/global/breakpoints';
+import breakpoints from '~tokens/global/breakpoints';
+import { getMediaQuery } from '~src/utils/getMediaQuery';
+import { isReactNative, isEmpty, getIn, makeSpace, makeSize } from '~utils';
 import type { Theme } from '~components/BladeProvider';
 
-// allows 'auto' as spacing value
-const getSpacingValue = <SpacingValue extends string | number | undefined>({
-  value,
-  theme,
-}: {
-  value: SpacingValue;
-  theme: Theme;
-}): string | undefined => {
-  if (value === 'auto') return 'auto';
-  /**
-   * NOTE: don't allow numbers once we have the scale for sizing
-   * DO NOT PUBLISH THIS COMPONENT PUBLICLY WITH THIS PIECE OF CODE
-   */
-  if (typeof value === 'number') {
-    return makeSpace(value);
+const getResponsiveValue = <T extends string | number | string[]>(
+  value: MakeValueResponsive<T> | undefined,
+  size: keyof Breakpoints = 'base',
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): any | undefined => {
+  if (value === undefined || value === null) {
+    return undefined;
   }
-  return value ? makeSpace(getIn(theme, value)) : undefined;
+
+  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value)) {
+    if (size === 'base') {
+      return value;
+    }
+
+    // Plain values will already get added in base styles so we don't have to repeat them in media queries
```

Comment:
This will explain it better. 
<img width="977" alt="image" src="https://user-images.githubusercontent.com/30949385/221789258-38872e30-a1db-484c-80a7-44a6288eba98.png">